# Week 2 Checkpoint: First Data Profile
**BAN 6003 Data Management and Analytics Integration**

Complete this notebook using the `employee_data.csv` file in the `data/` folder.

This is the independent checkpoint for Week 2. Use the functions practiced in the guided lab. You are not expected to clean the data yet. Your goal is to profile the data and report what you notice.


## Instructions

1. Run each cell from top to bottom.
2. Fill in all cells marked `# Your code here`.
3. Write short answers in the markdown response areas.
4. Save, commit, and push your notebook to GitHub.

You may use the Week 2 guided lab as a reference.


## Task 0: Import Pandas and Load the Data
Run the cell below.


In [9]:
import pandas as pd

df = pd.read_csv("../data/employee_data.csv")
df.head()


,employee_id,first_name,last_name,hire_date,department,salary,age,full_time
0,101,John,Smith,4/15/22,Marketing,85000,34.0,True
1,102,Jane,Doe,8/20/21,Sales,92000,45.0,True
2,103,Peter,Jones,1/2/22,HR,71000,29.0,True
3,104,Mary,Johnson,11/10/23,Engineering,115000,38.0,True
4,105,Mike,Brown,5/30/20,sales,89000,NaN,False


## Task 1: Basic Structure
Show the basic structure of the DataFrame.

Your code should show:
- the shape of the DataFrame;
- the column names;
- the first five rows;
- the last five rows.


In [10]:
print(df.shape)
print(df.columns)
display(df.head())
display(df.tail())




(13, 8)
Index(['employee_id', 'first_name', 'last_name', 'hire_date', 'department',
       'salary', 'age', 'full_time'],
      dtype='str')


,employee_id,first_name,last_name,hire_date,department,salary,age,full_time
0,101,John,Smith,4/15/22,Marketing,85000,34.0,True
1,102,Jane,Doe,8/20/21,Sales,92000,45.0,True
2,103,Peter,Jones,1/2/22,HR,71000,29.0,True
3,104,Mary,Johnson,11/10/23,Engineering,115000,38.0,True
4,105,Mike,Brown,5/30/20,sales,89000,NaN,False


,employee_id,first_name,last_name,hire_date,department,salary,age,full_time
8,102,Emily,White,1/15/24,Sales,91000,31.0,True
9,110,David,Miller,7/22/19,HR,68000,58.0,True
10,111,Sarah,Moore,3/12/23,Marketing,"$91,000",28.0,True
11,error,Kevin,Clark,2/5/24,Sales,88000,35.0,True
12,113,Brian,Hall,6/18/22,Engineering,5900000,999.0,True


### Task 1 Written Response
Write 1–2 sentences describing the size and general structure of the dataset.

**Your response:**  
[There appear to be 12 rows and 8 columns. Also, there is an error in the emp id column. There is also an "NaN" in the age column for Mike Brown in the 5th position. The data appear to be categorical in nature and also discrete and unranked ]


## Task 2: Data Types and Non-Missing Counts
Use Pandas to inspect data types and non-missing counts.

Your code should use:
- `info()`
- `dtypes`


In [15]:
df.info()
display(df.dtypes)

<class 'pandas.DataFrame'>
RangeIndex: 13 entries, 0 to 12
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   employee_id  13 non-null     str    
 1   first_name   13 non-null     str    
 2   last_name    13 non-null     str    
 3   hire_date    12 non-null     str    
 4   department   13 non-null     str    
 5   salary       12 non-null     str    
 6   age          12 non-null     float64
 7   full_time    12 non-null     object 
dtypes: float64(1), object(1), str(6)
memory usage: 964.0+ bytes


employee_id        str
first_name         str
last_name          str
hire_date          str
department         str
salary             str
age            float64
full_time       object
dtype: object

### Task 2 Written Response
Which columns, if any, have data types that might need to be investigated later? For example, is a number or date stored as text/object?

**Your response:**  
[In the emp there are duplicate 102's as well as an "error" for kevin Clark. Also it turns out that pandas uses the term object when there are either text, mixed data types, or python strings. Also, there are str listed for values which means there are text data like in salary column instead of just numbers]


## Task 3: Missing Values
Check missing values for all columns.

Your code should use `isna().sum()`.

Optional: also calculate missing percentages.


In [23]:
pd.DataFrame({
    'missing_count': df.isna().sum(),
    'missing_pct': (df.isna().mean() * 100).round(2)
})


,missing_count,missing_pct
employee_id,0,0.00
first_name,0,0.00
last_name,0,0.00
hire_date,1,7.69
department,0,0.00
salary,1,7.69
age,1,7.69
full_time,1,7.69


### Task 3 Written Response
Which columns have missing values? Which missing values seem most important to investigate later?

**Your response:**  
[Write your answer here]


## Task 4: Numerical Summary
Use `.describe()` to summarize numerical columns.

Then choose one numerical column and calculate at least two individual summary statistics such as mean, median, min, max, or standard deviation.


In [ ]:
# Your code here


### Task 4 Written Response
What do you notice from the numerical summaries? Mention at least one range, unusual value, or pattern.

**Your response:**  
[Write your answer here]


## Task 5: Categorical Summary
Use `.value_counts()` to summarize at least one categorical column.

Start with `department` if it exists in the dataset. Then choose one additional categorical column if available.


In [ ]:
# Your code here


### Task 5 Written Response
What are the most common categories? Do you notice anything that might need standardization or further investigation later?

**Your response:**  
[Write your answer here]


## Task 6: Duplicate Checks
Check for duplicate records and duplicate employee IDs.

Your code should check:
- fully duplicated rows;
- duplicate `employee_id` values;
- the most common `employee_id` values using `value_counts()`.


In [ ]:
# Your code here


### Task 6 Written Response
Are there duplicate employee IDs? If yes, what might that mean if `employee_id` is supposed to uniquely identify each employee?

**Your response:**  
[Write your answer here]


## Task 7: Overall Data Quality Observations
Write 3–5 bullet points summarizing the condition of the dataset.

Focus on what you found from profiling. Do **not** clean the data yet.

Examples:
- The dataset has ___ rows and ___ columns.
- The ___ column has missing values.
- The ___ column may have an incorrect data type.
- The ___ column has possible duplicate IDs.
- The ___ column has a range from ___ to ___.


**Your observations:**

1. [Observation 1]
2. [Observation 2]
3. [Observation 3]
4. [Optional Observation 4]
5. [Optional Observation 5]


## Submission Checklist
Before submitting, make sure:

- [ ] All code cells run without errors.
- [ ] You completed all tasks.
- [ ] You wrote short responses in the markdown cells.
- [ ] You saved the notebook.
- [ ] You committed and pushed your work to GitHub.

Use the terminal:

```bash
git add .
git commit -m "Completed Week 2 data profile checkpoint"
git push
```
